crates/algokit_transact/src/transactions/mod.rs

```diff
@@ -234,4 +234,99 @@ impl Transactions for &[Transaction] {
             })
             .collect())
     }
+
+    /// Encodes the supplied transactions to MsgPack format with the appropriate prefix (TX).
+    ///
+    /// This method performs canonical encoding and prepends the domain separation
+    ///
+    /// Use `encode_raw()` if you want to encode without the prefix.
+    ///
+    /// # Returns
+    /// The encoded bytes with prefix for the supplied transactions or an AlgoKitTransactError if serialization fails.
+    fn encode(self) -> Result<Vec<Vec<u8>>, AlgoKitTransactError> {
```

Comment:
I'm wondering if we need these impl functions. If someone needs this functionality it's easy enough to use `map` and that avoids the required allocation for returning the collected vec (I suspect in most cases they don't actually need to own the data)